testing/src/main/java/io/grpc/internal/testing/AbstractTransportTest.java

```diff
@@ -523,6 +540,18 @@ public void newStream_duringShutdown() throws Exception {
       inOrder.verify(clientStreamTracerFactory).newClientStreamTracer(any(Metadata.class));
     }
     ClientStreamListener mockClientStreamListener2 = mock(ClientStreamListener.class);
+    doAnswer(
+          new Answer<Void>() {
+            @Override
+            public Void answer(InvocationOnMock invocation) throws Throwable {
+              StreamListener.MessageProducer producer =
+                  (StreamListener.MessageProducer) invocation.getArguments()[0];
+              while (producer.next() != null) {}
```

Comment:
I'd really hope we'd close the messages properly everywhere in AbstractTransportTest.